From 9c042b9a38a522e8e977ea4845dfc98608e9c6cb Mon Sep 17 00:00:00 2001
From: "#Henkate @xda" <robertpopescu95@yahoo.com>
Date: Fri, 11 Aug 2017 19:26:12 +0300
Subject: [PATCH] dont build CMUpdater - this is an unofficial build

---
 config/common.mk | 1 -
 1 file changed, 1 deletion(-)

diff --git a/config/common.mk b/config/common.mk
index 56712833..b4269822 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -121,7 +121,6 @@ PRODUCT_PACKAGES += \
     CMFileManager \
     Eleven \
     LockClock \
-    CMUpdater \
     CyanogenSetupWizard \
     CMSettingsProvider \
     ExactCalculator \
