From eb7852931ed2aa7129b266fe5b48b514d81f36a9 Mon Sep 17 00:00:00 2001
From: "#Henkate @xda" <robertpopescu95@yahoo.com>
Date: Sat, 16 Sep 2017 18:55:44 +0300
Subject: [PATCH 1/2] add msm8953 support and updates from LOS [1/2]

---
 core/qcom_target.mk | 69 +++++++++++++++++++++++++----------------------------
 1 file changed, 32 insertions(+), 37 deletions(-)

diff --git a/core/qcom_target.mk b/core/qcom_target.mk
index 161241b3ea..40b1775f01 100644
--- a/core/qcom_target.mk
+++ b/core/qcom_target.mk
@@ -1,5 +1,8 @@
 # Target-specific configuration
 
+# Bring in Qualcomm helper macros
+include $(BUILD_SYSTEM)/qcom_utils.mk
+
 # Populate the qcom hardware variants in the project pathmap.
 define ril-set-path-variant
 $(call project-set-path-variant,ril,TARGET_RIL_VARIANT,hardware/$(1))
@@ -22,21 +25,28 @@ $(call project-set-path,qcom-$(2),$(strip $(path)))
 endef
 
 ifeq ($(BOARD_USES_QCOM_HARDWARE),true)
+    B_FAMILY := msm8226 msm8610 msm8974
+    B64_FAMILY := msm8992 msm8994
+    BR_FAMILY := msm8909 msm8916
+    UM_FAMILY := msm8937 msm8953
 
     qcom_flags := -DQCOM_HARDWARE
     qcom_flags += -DQCOM_BSP
     qcom_flags += -DQTI_BSP
 
+    BOARD_USES_ADRENO := true
+
     TARGET_USES_QCOM_BSP := true
 
     # Tell HALs that we're compiling an AOSP build with an in-line kernel
     TARGET_COMPILE_WITH_MSM_KERNEL := true
 
-    ifneq ($(filter msm7x30 msm8660 msm8960,$(TARGET_BOARD_PLATFORM)),)
+    ifneq ($(filter msm7x27a msm7x30 msm8660 msm8960,$(TARGET_BOARD_PLATFORM)),)
         # Enable legacy graphics functions
         qcom_flags += -DQCOM_BSP_LEGACY
         # Enable legacy audio functions
         ifeq ($(BOARD_USES_LEGACY_ALSA_AUDIO),true)
+            USE_CUSTOM_AUDIO_POLICY := 1
             qcom_flags += -DLEGACY_ALSA_AUDIO
         endif
     endif
@@ -52,46 +62,32 @@ ifeq ($(BOARD_USES_QCOM_HARDWARE),true)
     2ND_CLANG_TARGET_GLOBAL_CFLAGS += $(qcom_flags)
     2ND_CLANG_TARGET_GLOBAL_CPPFLAGS += $(qcom_flags)
 
-    ifeq ($(QCOM_HARDWARE_VARIANT),)
-        ifneq ($(filter msm8610 msm8226 msm8974,$(TARGET_BOARD_PLATFORM)),)
-            QCOM_HARDWARE_VARIANT := msm8974
-        else
-        ifneq ($(filter msm8909 msm8916,$(TARGET_BOARD_PLATFORM)),)
-            QCOM_HARDWARE_VARIANT := msm8916
-        else
-        ifneq ($(filter msm8992 msm8994,$(TARGET_BOARD_PLATFORM)),)
-            QCOM_HARDWARE_VARIANT := msm8994
-        else
-            QCOM_HARDWARE_VARIANT := $(TARGET_BOARD_PLATFORM)
-        endif
-        endif
-        endif
+    ifeq ($(call is-board-platform-in-list, $(B_FAMILY)),true)
+        MSM_VIDC_TARGET_LIST := $(B_FAMILY)
+        QCOM_HARDWARE_VARIANT := msm8974
+    else
+    ifeq ($(call is-board-platform-in-list, $(B64_FAMILY)),true)
+        MSM_VIDC_TARGET_LIST := $(B64_FAMILY)
+        QCOM_HARDWARE_VARIANT := msm8994
+    else
+    ifeq ($(call is-board-platform-in-list, $(BR_FAMILY)),true)
+        MSM_VIDC_TARGET_LIST := $(BR_FAMILY)
+        QCOM_HARDWARE_VARIANT := msm8916
+    else
+    ifeq ($(call is-board-platform-in-list, $(UM_FAMILY)),true)
+        MSM_VIDC_TARGET_LIST := $(UM_FAMILY)
+        QCOM_HARDWARE_VARIANT := msm8937
+    else
+        MSM_VIDC_TARGET_LIST := $(TARGET_BOARD_PLATFORM)
+        QCOM_HARDWARE_VARIANT := $(TARGET_BOARD_PLATFORM)
+    endif
+    endif
+    endif
     endif
 
-# HACK: check to see if build uses standard QC HAL paths by checking for CM path structure
-AOSP_VARIANT_MAKEFILE := $(wildcard hardware/qcom/audio/default/Android.mk)
-ifeq ("$(AOSP_VARIANT_MAKEFILE)","")
-$(call project-set-path,qcom-audio,hardware/qcom/audio)
-$(call project-set-path,qcom-display,hardware/qcom/display)
-$(call project-set-path,qcom-media,hardware/qcom/media)
-$(call set-device-specific-path,CAMERA,camera,hardware/qcom/camera)
-$(call set-device-specific-path,GPS,gps,hardware/qcom/gps)
-$(call set-device-specific-path,SENSORS,sensors,hardware/qcom/sensors)
-$(call set-device-specific-path,LOC_API,loc-api,vendor/qcom/opensource/location)
-$(call set-device-specific-path,DATASERVICES,dataservices,vendor/qcom/opensource/dataservices)
-$(call project-set-path,ril,hardware/ril)
-$(call project-set-path,wlan,hardware/qcom/wlan)
-$(call project-set-path,bt-vendor,hardware/qcom/bt)
-else
 $(call project-set-path,qcom-audio,hardware/qcom/audio-caf/$(QCOM_HARDWARE_VARIANT))
-
-ifeq ($(SONY_BF64_KERNEL_VARIANT),true)
-$(call project-set-path,qcom-display,hardware/qcom/display-caf/sony)
-$(call project-set-path,qcom-media,hardware/qcom/media-caf/sony)
-else
 $(call project-set-path,qcom-display,hardware/qcom/display-caf/$(QCOM_HARDWARE_VARIANT))
 $(call project-set-path,qcom-media,hardware/qcom/media-caf/$(QCOM_HARDWARE_VARIANT))
-endif
 
 $(call set-device-specific-path,CAMERA,camera,hardware/qcom/camera)
 $(call set-device-specific-path,GPS,gps,hardware/qcom/gps)
@@ -102,7 +98,6 @@ $(call set-device-specific-path,DATASERVICES,dataservices,vendor/qcom/opensource
 $(call ril-set-path-variant,ril)
 $(call wlan-set-path-variant,wlan-caf)
 $(call bt-vendor-set-path-variant,bt-caf)
-endif # AOSP_VARIANT_MAKEFILE
 
 else
 

From a5f3410c37ccf25cc44b6fa83148d2a235557ad4 Mon Sep 17 00:00:00 2001
From: "#Henkate @xda" <robertpopescu95@yahoo.com>
Date: Sat, 16 Sep 2017 18:56:33 +0300
Subject: [PATCH 2/2] add msm8953 support and updates from LOS [2/2]

---
 core/qcom_utils.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/qcom_utils.mk b/core/qcom_utils.mk
index 5e8a4a2a2c..50e0b4e04b 100755
--- a/core/qcom_utils.mk
+++ b/core/qcom_utils.mk
@@ -1,5 +1,6 @@
 # Board platforms lists to be used for
 # TARGET_BOARD_PLATFORM specific featurization
+QCOM_BOARD_PLATFORMS += msm7x27a
 QCOM_BOARD_PLATFORMS += msm7x30
 QCOM_BOARD_PLATFORMS += msm8226
 QCOM_BOARD_PLATFORMS += msm8610
@@ -9,7 +10,9 @@ QCOM_BOARD_PLATFORMS += msm8916
 QCOM_BOARD_PLATFORMS += msm8960
 QCOM_BOARD_PLATFORMS += msm8974
 QCOM_BOARD_PLATFORMS += mpq8092
+QCOM_BOARD_PLATFORMS += msm8937
 QCOM_BOARD_PLATFORMS += msm8952
+QCOM_BOARD_PLATFORMS += msm8953
 QCOM_BOARD_PLATFORMS += msm8992
 QCOM_BOARD_PLATFORMS += msm8994
 QCOM_BOARD_PLATFORMS += msm8996
